// src/firebase.js
import { initializeApp } from 'firebase/app'; // Importamos Firebase core
import { getFirestore } from 'firebase/firestore'; // Importamos Firestore
import { getAuth } from 'firebase/auth'; // Importamos Firebase Auth
import { getStorage } from 'firebase/storage'; // Importamos Firebase Storage

// Configuración de Firebase
const firebaseConfig = {
    apiKey: "AIzaSyBu-6yCT5WLhut5ckSco8KVmr_cc0bSLxQ",
    authDomain: "tienda-productos-20b99.firebaseapp.com",
    projectId: "tienda-productos-20b99",
    storageBucket: "tienda-productos-20b99.appspot.com", // Corrige el URL del storageBucket
    messagingSenderId: "628605223530",
    appId: "1:628605223530:web:b4b6f1711b255b7caa4013"
};

// Inicializamos Firebase
const app = initializeApp(firebaseConfig);
const firestore = getFirestore(app); // Obtenemos la instancia de Firestore
const auth = getAuth(app); // Obtenemos la instancia de Auth
const storage = getStorage(app); // Obtenemos la instancia de Storage

//Registro de Usuario
function register(){
    const email = document.getElementById("email").value;
    const password = document.getElementById("password").value;

    auth.createUserEmailAndPassword(email, password)
    .then((userCrefential) => {
        alert("User registered successfully!");
        document.getElementById("authForm").reset();
        document.getElementById("authSection").style.display = "none";
        document.getElementById("uploadSection").style.display = "block";
    })
    .catch((error) => {
        alert(error.message);
    });
}

//Login de Usuario
function login(){
    const email = document.getElementById("email").value;
    const password = document.getElementById("password").value;

    auth.signInWithEmailAndPassword(email, password)
    .then((userCrefential) => {
        alert("User logged in successfully!");
        document.getElementById("authForm").reset();
        document.getElementById("authSection").style.display = "none";
        document.getElementById("uploadSection").style.display = "block";
    })
    .catch((error) => {
        alert(error.message);
    });
}

//Subida de archivo
function uploadFile() {
    const file = document.getElementById("fileInput").files(0);
    if (!file) {
        alert("Please choose a file first.");
        return;
    }

    const storageRef = storage.ref("upload/" + file.name);
    const uploadTask =storageRef.put(file);

    uploadTask.on(
        "State_changed",
        (snapshot) => {
            const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
            console.log("Upload is" + progress +"% done");
        },
        (error) => {
            alert(error.message);
        },
        () => {
            uploadTask.snapshot.ref.getDownloadURL().then((downloadURL) => {
                alert("File uploaded successfully! File URL: " + downloadURL);
            });
        }
    );

}

//Ocultar sección de subida si no hay usuario autenticado
auth.onAuthStateChanged((user) =>{
    if (user) {
        document.getElementById("authSection").style.display = "none";
        document.getElementById("uploadSection").style.display = "block";
    } else {
        document.getElementById("authSection").style.display = "none";
        document.getElementById("uploadSection").style.display = "block";
    }
});


export { firestore, auth, storage }; // Exportamos nuestras instancias configuradas.
